java/ql/lib/semmle/code/java/dataflow/FlowSources.qll

```diff
@@ -175,14 +206,29 @@ abstract class UserInput extends DataFlow::Node { }
 private class RemoteUserInput extends UserInput instanceof RemoteFlowSource { }
 
 /** A node with input that may be controlled by a local user. */
-abstract class LocalUserInput extends UserInput { }
+abstract class LocalUserInput extends UserInput, SourceNode {
+  override string getThreatModel() { result = "local" }
+}
 
 /**
  * A node with input from the local environment, such as files, standard in,
  * environment variables, and main method parameters.
  */
-class EnvInput extends LocalUserInput {
+private class EnvInput extends LocalUserInput {
```

Comment:
You're making some classes private, which maybe ought to go through deprecation. It should be simple to use a slightly tweaked private name and then expose a deprecated alias. It seems reasonable to expect that there could be custom queries out there that explicitly reference some of these sources, so they'd need some guidance on how to convert to the new setup.

Reply:
That is a valid point. I will re-expose a re-implementation of `EnvInput` and deprecate it. Furthermore, also add some hints, how to add implement something that is equivalent. If it turns out that it is really needed by customers that they have this exact class of nodes we can consider removing the deprecation again.